src/marten/db/query/sql/raw_statement_node.cr

```diff
@@ -0,0 +1,70 @@
+require "./concerns/sanitizer"
+
+module Marten
+  module DB
+    module Query
+      module SQL
+        class RawStatementNode < PredicateNode
+          include Sanitizer
+
+          getter statement
+          getter params
+
+          def initialize(
+            @statement : String,
+            @params = [] of ::DB::Any,
+            @children = [] of PredicateNode,
+            @connector = SQL::PredicateConnector::AND,
+            @negated = false
+          )
+            @predicates = [] of Predicate::Base
+          end
+
+          def initialize(
+            @statement : String,
+            @params : Array(::DB::Any) | Hash(String, ::DB::Any),
+            @children : Array(PredicateNode),
+            @connector : PredicateConnector,
+            @negated : Bool,
+            @predicates : Array(Predicate::Base)
+          )
+          end
+
+          def ==(other : RawStatementNode)
+            (
+              (other.statement == statement) &&
+                (other.params == params) &&
+                (other.children == children) &&
+                (other.connector == connector) &&
+                (other.negated == negated)
+            )
+          end
+
+          def clone
+            RawStatementNode.new(
+              statement: @statement,
+              params: params,
+              children: @children.map(&.clone),
+              connector: @connector,
+              negated: @negated,
+              predicates: @predicates.dup
+            )
+          end
+
+          def to_sql(connection : Connection::Base)
+            # Escape % characters to be not be considered
+            # during formatting process
+            @statement = @statement.gsub("%", "%%")
```

Comment:
I may be missing something, but why is this needed? Afaik, this is not something that we are doing when preparing raw SQL queries presently.

Reply:
Given I have a filter with LIKE this is necessary. Post.all.find("name LIKE 'C%'") won't be correctly formatted ( [`where_clause_and_parameters`](https://github.com/martenframework/marten/blob/main/src/marten/db/query/sql/query.cr#L1117) and will throw a **Malformed format string** error

Reply:
Gotcha, that makes sense. Can we add a spec to verify that this behavior is in place? 🙏